### ISSUE TITLE
Snapshot stop working

### ISSUE THREAD

**Issue by PatrikMihalik** [AdyRock/com.onvif#58]
Hello
I have problem with onvif app.
Snapshot stop working after update,till this time working good. I see.snapshot in import device and sending.via flow working good. But now everything stop working. 

f392410e-9281-4779-9b63-3429bcbe91c3

Please look into report where is the problem.
Thank you.

**Comment by PatrikMihalik** [comment]
After hard reset and adding device again,snapshot start working but motion alarm in homey not working,I see camera time,but motion is without notification 2 days.
![Screenshot_2023-08-24-21-02-24-392_app homey](https://github.com/AdyRock/com.onvif/assets/129721129/4ff48360-e722-4208-800b-3f9162404af8)

**Comment by AdyRock** [comment]
Did the motion work before the update?

**Comment by PatrikMihalik** [comment]
Yes it is worked, I have flow with sending image via messenger to chatbot  and 2days nothing happens. But original IMOU LIFE app from manufacturer works and sending notification.

**Comment by AdyRock** [comment]
Could you set the log to Basic Information, then trigger motion on the camera and then send the log.

**Comment by PatrikMihalik** [comment]
Log was sent to you. I switch the motion off and on.

**Comment by AdyRock** [comment]
Ok, I can see lots of errors, so could you set the log level to Full and do the same. Hopefully I will be able to see the details of the cause of the errors.

**Comment by PatrikMihalik** [comment]
OK, log was sent. My camera is IMOU (dahua) bullet. technical name IPC-F42. Please look on it,maybe I set something wrong.  I dont know. :-(

**Comment by PatrikMihalik** [comment]
Hello Ady,

anything new in case with my camera?
After re-adding camera snapshot start works,but the motion trigger didnt work yet.

**Comment by PatrikMihalik** [comment]
Hello Ady anything new in this case???

**Comment by AdyRock** [comment]
Looking at the log, it seems the camera is having some issues:

`_request response (192.168.0.116): <?xml version="1.0" encoding="utf-8" standalone="yes" ?><s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:sc="http://www.w3.org/2003/05/soap-encoding" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:wsrf-bf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:d="http://schemas.xmlsoap.org/ws/2005/04/discovery"><s:Header><wsa5:Action>http://www.w3.org/2005/08/addressing/soap/fault</wsa5:Action></s:Header><s:Body><s:Fault><s:Code><s:Value>s:Sender</s:Value></s:Code><s:Reason><s:Text xml:lang="en">Subscribe Creation Failed</s:Text></s:Reason><s:Detail><wsnt:SubscribeCreationFailedFault><wsrf-bf:ErrorCode dialect="http://docs.oasis-open.org/wsrf/r-2"></wsrf-bf:ErrorCode><wsrf-bf:Description>The NotificationProducer failed to process the Subscribe message.The NotificationProducer SHOULD use a more specific fault message if possible.The NotificationProducer MAY include a hint in the fault message indicating why it failed to process the Subscribe message.</wsrf-bf:Description></wsnt:SubscribeCreationFailedFault></s:Detail></s:Fault></s:Body></s:Envelope>`

When the app tries to subscribe to events it gets some strange errors back that I haven't seen before.
Have you checked to see if there are any firmware updates for the camera?

**Comment by PatrikMihalik** [comment]
Hello Ady,

Can I help you with solving my problem with dahuas camera?

**Comment by AdyRock** [comment]
Could you open the app settings page, select the Diagnostics Log tab and set the Log Level to Detailed Information.
Then grab a snapshot, return to the Log and tap on Send Log.
The Homey Diagnostics Log doesn't contain any useful information.

**Comment by PatrikMihalik** [comment]
Everything done like you describe.

**Comment by AdyRock** [comment]
According to the Log the image fetch was successful. What makes you say it's not working?

**Comment by PatrikMihalik** [comment]
I look on file,it is without error,instead of that,camera didnt send notification about motion snapshot or motion event in onvif. But in standard imou app everything working OK. If I manually start flow to send snapshot via chatbot everything is working,but motion in onvif not working correctly.
![Screenshot_2023-11-12-14-46-12-658_app homey-edit](https://github.com/AdyRock/com.onvif/assets/129721129/0c13ac30-847c-4f2a-bf2f-44d48e5db70c)

**Comment by PatrikMihalik** [comment]
My setting in onvif,please check it if its correct set.
![Screenshot_2023-11-12-14-50-02-088_app homey](https://github.com/AdyRock/com.onvif/assets/129721129/468a280d-646c-43a0-bf20-1e0015361d3e)

**Comment by AdyRock** [comment]
Ah, OK so it's the motion detection that's not working?

If so could you setup the log as you did before, but this time trigger the camera motion so I can see what messages it sent.

**Comment by PatrikMihalik** [comment]
Ok I sent two reports,in first i forget switch on motion detection. But in last report I switch it on. In onvif camera didnt trigger motion.

**Comment by AdyRock** [comment]
I can see the motion event coming in a being processed. The only possible odd thing is it's showing it's ignored it. I will investigate the circumstances for that tomorrow, but as far as I can remember that is when the event is triggered while the app is still timing out a previous trigger. 
I see in the settings you have single set to true which would ignore events if they occur less than 2 seconds apart.

**Comment by AdyRock** [comment]
The only other thing to check is the "ONVIF Motion Notification to use" and make sure that is set to Motion (isMotion)

**Comment by PatrikMihalik** [comment]
I try to change single set to false,but no motion was triggered. Onvif is without motion detection.

Notification to use is set to motion is motion.

**Comment by AdyRock** [comment]
Doh, I just spotted that the "ONVIF Motion Notification to use" should be set to "MotionAlarm". I can't believe I missed that before.

**Comment by PatrikMihalik** [comment]
Great now everything working :-) Thank you :-)
